heron/healthm/src/java/com/twitter/heron/healthmgr/HealthManager.java

```diff
@@ -0,0 +1,355 @@
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+import com.twitter.heron.healthmgr.common.HealthMgrConstants;
+import com.twitter.heron.healthmgr.common.PackingPlanProvider;
+import com.twitter.heron.healthmgr.sensors.TrackerMetricsProvider;
+import com.twitter.heron.scheduler.client.ISchedulerClient;
+import com.twitter.heron.scheduler.client.SchedulerClientFactory;
+import com.twitter.heron.spi.common.Config;
+import com.twitter.heron.spi.common.ConfigLoader;
+import com.twitter.heron.spi.common.Context;
+import com.twitter.heron.spi.common.Key;
+import com.twitter.heron.spi.statemgr.IStateManager;
+import com.twitter.heron.spi.statemgr.SchedulerStateManagerAdaptor;
+import com.twitter.heron.spi.utils.ReflectionUtils;
+
+/**
+ * e.g. options
+ * -d ~/.heron -p ~/.heron/conf/local -c local -e default -r userName -t AckingTopology
+ */
+public class HealthManager {
+  private static final Logger LOG = Logger.getLogger(HealthManager.class.getName());
+  private final Config config;
+  private final String trackerURL;
+
+  private Config runtime;
+  private Injector injector;
+  private SchedulerStateManagerAdaptor stateMgrAdaptor;
+  private ISchedulerClient schedulerClient;
+
+  private List<IHealthPolicy> healthPolicies = new ArrayList<>();
+  private HealthPolicyConfigReader policyConfigReader;
+
+  public HealthManager(Config config, String trackerUrl) {
+    this.config = config;
+    this.trackerURL = trackerUrl;
+  }
+
+  public static void main(String[] args) throws Exception {
+    CommandLineParser parser = new DefaultParser();
+    Options slaManagerCliOptions = constructCliOptions();
+
+    // parse the help options first.
+    Options helpOptions = constructHelpOptions();
+    CommandLine cmd = parser.parse(helpOptions, args, true);
+    if (cmd.hasOption("h")) {
+      usage(slaManagerCliOptions);
+      return;
+    }
+
+    try {
+      cmd = parser.parse(slaManagerCliOptions, args);
+    } catch (ParseException e) {
+      usage(slaManagerCliOptions);
+      throw new RuntimeException("Error parsing command line options: ", e);
+    }
+
+    String cluster = cmd.getOptionValue("cluster");
+    String role = cmd.getOptionValue("role");
+    String environ = cmd.getOptionValue("environment");
+    String heronHome = cmd.getOptionValue("heron_home");
+    String configPath = cmd.getOptionValue("config_path");
+    String overrideConfigFile = cmd.getOptionValue("override_config_file");
+    String releaseFile = cmd.getOptionValue("release_file");
+    String topologyName = cmd.getOptionValue("topology_name");
+    String trackerUrl = cmd.getOptionValue("trackerURL", "http://localhost:8888");
+    Boolean verbose = cmd.hasOption("verbose");
+
+    // build the final config by expanding all the variables
+    Config config = Config.toLocalMode(Config.newBuilder()
+        .putAll(ConfigLoader.loadConfig(heronHome, configPath, releaseFile, overrideConfigFile))
```

Comment:
As per the javadocs of `loadClusterConfig`, the method assumes default home and config path: `Loads raw configurations using the default configured heronHome and configPath on the cluster`. This may not be the case here. The method is only useful for `scheduler` service only.